### PR TITLE
VFD 47 - Inital Commit

### PR DESCRIPTION
**Pull request by adriansimionescuRebase**
This PR the following fetures and functionalities:
* Users API + Integration Tests
* Swagger UI available in dev configurations
* Pulumi Infrastructure definitions for AWS resources
* AWS Resources: Using infrastructure projects VPC Pulumi definitions, RDS database + lambda configurations to support VPC attachment
* Lambda function is configured with Pulumi to be inside a VPC to access the RDS database securely and still be available publicly
* Docker support for local dev situations, the ability to start the API with a database in docker

Users API business requirements:
* The ability to use Testbed authentication to identify a user
* Create a new user into the API DB if the Testbed auth information was ok and it didn't exist before
* For the auth user the ability to store personal information
* For the auth user the ability to store job search parameters

Things to know:
* Support only one development configuration
* Missing secrets management at the moment
* Pulumi unit tests missing because no clear way to unit tests stack refences at the moment
* The Users API uses Vertical Slice Architecture style pattern with CQRS (structured around features and capabilities)

### CODE REVIEW

File: VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using System.Globalization;
+
+namespace VirtualFinland.UserAPI.Exceptions;
+
+public class NotAuthorizedExpception : Exception
```
Comment (lassipatanen) [inline VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs:5]: Typo luokan nimessä

File: VirtualFinland.UsersAPI.sln [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿
```
Comment (lassipatanen) [inline VirtualFinland.UsersAPI.sln:1]: .DS_Store voisi lisätä gitignoreen

**Review by lassipatanen — APPROVED** [review]
Hyvää duunia!

File: VirtualFinland.UsersAPI.sln (hunk 2 above)
Comment (adriansimionescuRebase) [reply VirtualFinland.UsersAPI.sln:1]: Thx hyvä huomio, poistin.

File: VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs (hunk 1 above)
Comment (adriansimionescuRebase) [reply VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs:5]: Moikka, en löydä typöä, sorry en vain näe, sokea piste:). 

Otin tässä taptauksessa mallia MS:n HttpStatusCode.Unauthorized http status enumista. Tarkoitatko nimeksi esim UnauthorizedException?

Comment (lassipatanen) [reply VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs:5]: tuossa välissä on yksi p liikaa

NotAuthorizedExpception -> NotAuthorizedException

Comment (adriansimionescuRebase) [reply VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Exceptions/NotAuthorizedExpception.cs:5]: DANG :D niin on, en huomannut ollenkaan LOL, thx korjaan :).

**Pull request merged**